lib/livebook/teams/requests.ex

```diff
@@ -86,6 +87,69 @@ defmodule Livebook.Teams.Requests do
     delete("/api/v1/org/secrets", params, headers)
   end
 
+  @doc """
+  Send a request to Livebook Team API to create a file system.
+  """
+  @spec create_file_system(Team.t(), FileSystem.t()) ::
+          {:ok, map()} | {:error, map() | String.t()} | {:transport_error, String.t()}
+  def create_file_system(team, file_system) do
+    {secret_key, sign_secret} = Teams.derive_keys(team.teams_key)
+
+    headers = auth_headers(team)
+    {type, name} = FileSystem.resource_identifier(file_system)
```

Comment:
`resource_identifier` should be opaque. We can either have a private function that matches on the struct and extracts this info, or extend the protocol with `def metadata(file_system) -> %{type: atom(), name: String.t()}`.

Reply:
I'm more into extending the protocol, it would be more easy to implement in every file system

Reply:
In this case, wouldn't be better to only return `{type :: atom(), name :: String.t()}`

Reply:
Could be either, since we will know the exact shape, but I would do map just because it's easier to extend.

Reply:
In this case, what about the `value` field to make it easy to generate the params to Teams API?

Reply:
Actually, we could have a `Metadata` struct to standardize the struct that we could use to communicate with Teams API

Reply:
It's fine to just have it in the typespec, and literally do `%{type: type, name: name} = FileSystem.metadata(file_system)` here.

Reply:
What if it returns `{type, name_field}`? This way, we can use this information to map the errors to the `id_field`. In this case, the `name_field` would be in bucket_url and we would map errors from name and value back to the `name_field`.

Reply:
Works for me, I'd still return a map though :)

Reply:
Map is good then!

Reply:
We settled on `%{name: ..., error_field: error_field} = LB.FileSystem.external_metadata(filesystem)`